pkg/detectors/gitlab/v1/gitlab_integration_test.go

```diff
@@ -297,6 +297,100 @@ func TestGitlab_FromChunk_WithV2Secrets(t *testing.T) {
 	}
 }
 
+// This test ensures gitlab v1 detector does not work on gitlab v3 secrets
+func TestGitlab_FromChunk_WithV3Secrets(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	defer cancel()
+	testSecrets, err := common.GetSecret(ctx, "trufflehog-testing", "detectors6")
+	if err != nil {
+		t.Fatalf("could not get test secrets from GCP: %s", err)
+	}
+	secret := testSecrets.MustGetField("GITLABV3")
+	secretInactive := testSecrets.MustGetField("GITLABV3_INACTIVE")
+	type args struct {
+		ctx    context.Context
+		data   []byte
+		verify bool
+	}
+	tests := []struct {
+		name                string
+		s                   Scanner
+		args                args
+		want                []detectors.Result
+		wantErr             bool
+		wantVerificationErr bool
+	}{
+		{
+			name: "verified v3 secret, not found",
+			s:    Scanner{},
+			args: args{
+				ctx:    context.Background(),
+				data:   []byte(fmt.Sprintf("You can find a gitlab super secret %s within", secret)),
+				verify: true,
+			},
+			want:    nil,
+			wantErr: false,
+		},
+		{
+			name: "verified v3 secret, not found",
+			s:    Scanner{},
+			args: args{
+				ctx:    context.Background(),
+				data:   []byte(fmt.Sprintf("gitlab %s", secret)),
+				verify: true,
+			},
+			want:    nil,
+			wantErr: false,
+		},
```

Comment:
Could you clarify the difference between these two? Does changing the length of the data influence the detection behavior in any way?

Reply:
Hmm, to be honest I pretty much copy pasted these tests. The only difference I can think of is the difference in proximity of the secret to the keyword.

Reply:
It is for `v1`. The purpose of this test is basically to ensure our `v1` detector does not detect `v3` secrets.